### ISSUE TITLE
protoc-gen-tstypes: Nested enums duplication identifier & namespace issue

### ISSUE THREAD

**Issue by mousavian** [tmc/grpcutils#20]
Hi Travis,
Assume we have a message definition like this:
```proto
syntax = "proto3";

package nested;

message Notification {
  enum Type {
    UNSPECIFIED = 0;
    TEXT = 1;
    VIDEO = 2;
    AUDIO = 3;
  }

  Type message_type = 1;
  string content = 2;
}

message Tweet {
  enum Type {
    UNSPECIFIED = 0;
    ORIGINAL = 1;
    RETWEET = 2;
  }

  Type tweet_type = 1;
  string content = 2;
}
```

When we compile, there will be two Enums under same namespace and `tsc` check throws this error:
```
TS2300: Duplicate identifier 'UNSPECIFIED'.
UNSPECIFIED = "UNSPECIFIED",
~~~~~~~~~~~
```

Output:
```typescript
declare namespace nested {
    export enum Type {
        UNSPECIFIED = "UNSPECIFIED",
        TEXT = "TEXT",
        VIDEO = "VIDEO",
        AUDIO = "AUDIO",
    }
    export interface Notification {
        messageType?: Type;
        content?: string;
    }
    export enum Type {
        UNSPECIFIED = "UNSPECIFIED",
        ORIGINAL = "ORIGINAL",
        RETWEET = "RETWEET",
    }
    export interface Tweet {
        tweetType?: Type;
        content?: string;
    }
}
```

Do you any idea how this can get resolved without renaming enum names?

**Comment by Akiq2016** [comment]
input
```proto
syntax = "proto3";

package nested;

message Notification {
  enum Type {
    UNSPECIFIED = 0;
    TEXT = 1;
    VIDEO = 2;
    AUDIO = 3;
  }

  Type message_type = 1;
  string content = 2;
}

message Tweet {
  enum Type {
    UNSPECIFIED = 0;
    ORIGINAL = 1;
    RETWEET = 2;
  }

  Type tweet_type = 1;
  string content = 2;
}
```

the expected output should be like this ?
```ts
declare namespace nested {
    export enum Notification_Type {
        UNSPECIFIED = "UNSPECIFIED",
        TEXT = "TEXT",
        VIDEO = "VIDEO",
        AUDIO = "AUDIO",
    }
    export interface Notification {
        messageType?: string;
        content?: string;
    }
    export enum Tweet_Type {
        UNSPECIFIED = "UNSPECIFIED",
        ORIGINAL = "ORIGINAL",
        RETWEET = "RETWEET",
    }
    export interface Tweet {
        tweetType?: string;
        content?: string;
    }
}
```
@tmc

**Comment by tmc** [comment]
This is resolved by #24
